pkg/reconciler/apis/apiexport/apiexport_controller.go

```diff
@@ -205,20 +206,25 @@ func (c *controller) enqueueAllAPIExports(clusterWorkspaceShard interface{}) {
 }
 
 func (c *controller) enqueueSecret(obj interface{}) {
-	secretKey, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
+	secretKey, err := kcpcache.DeletionHandlingMetaClusterNamespaceKeyFunc(obj)
 	if err != nil {
 		runtime.HandleError(err)
 		return
 	}
 
-	apiExportKeys, err := c.apiExportIndexer.IndexKeys(indexers.APIExportBySecret, secretKey)
+	apiExports, err := c.apiExportIndexer.ByIndex(indexers.APIExportBySecret, secretKey)
 	if err != nil {
 		runtime.HandleError(err)
 		return
 	}
 
 	logger := logging.WithObject(logging.WithReconciler(klog.Background(), controllerName), obj.(*corev1.Secret))
-	for _, key := range apiExportKeys {
+	for _, apiExport := range apiExports {
+		key, err := kcpcache.DeletionHandlingMetaClusterNamespaceKeyFunc(apiExport)
```

Comment:
deletion handling is strange here.

Reply:
Similar to our conversation - if it's always correct to use `DeletionHandling`, let's just do it. It degenerates gracefully to the right thing and there's less mental overhead to figuring out when it is and when it is not necessary.

Reply:
There's no harm in doing it. But it's only strictly necessary when processing objects you get directly from AddFunc/UpdateFunc/DeleteFunc. Here, the objects are coming directly from the indexer, so you know they weren't deleted.

Reply:
It's a fairly nuanced point and unless we think there's real value from not just using `DeletionHandling` everywhere IMHO it's going to be easier for us to be provably correct & doing the right thing if we just use it everywhere (I *think* you agree, since you suggested we drop the `DeletionHandling` option from our keyfuncs and make it default everywhere)

Reply:
Unless/until we see any performance issues, we know we can't mess up if we use the deletion handling variant 100% of the time, so yes I am in favor of that.